mapstructure_test.go

```diff
@@ -2316,17 +2351,12 @@ func TestDecode_StructTaggedWithOmitempty_KeepNonEmptyValues(t *testing.T) {
 		VisibleMapField:    nil,
 		OmitMapField:       map[string]interface{}{"k": "v"},
 		NestedField:        nil,
-		OmitNestedField:    &Nested{},
+		OmitNestedField:    nil,
```

Comment:
Was this change intended? Looks a bit suspicious to me.

Reply:
Yes, this is intended. This PR change the behavior of struct pointer to allow decoding them. This means that &Nested{} is no longer considered `nil` once decoded; it will be a empty `map[string]interface{}`. Since this is not what is tested here, a separate unit test (`TestNestedTypeStructPointer`) was added just for this and this test was modified to pass `nil` so that it is not overloaded.